Add function to normalize Alfred input

kiliankoe opened the pull request:
For problems regarding http://www.alfredforum.com/topic/2015-encoding-issue/

I've previously been using something directly in go, but that was far from perfect... And iconv should be installed on every Mac anyways.

kiliankoe committed:
Add function to normalize Alfred input

BenchR267 commented:
Thanks, looks like a great addition! But I would like to see that integrated in the GetArguments logic. What do you think about adding a new function NormalizedArguments() which gets the arguments and normalizes them before returning?

kiliankoe committed:
Add func NormalizedArgument()

also change return type of Argument() to a single string,
which is what Alfred appears to be doing anyways.

Commit:
Arguments is an array of string

BenchR267 commented:
If you add yourself and your change to the CHANGELOG.md, I would love to merge this :)

kiliankoe committed:
Immediately return on error in NormalizedArgs

kiliankoe committed:
Add re-normalizing to CHANGELOG

BenchR267 commented on response.go:
```diff
@@ -11,6 +11,21 @@ func Arguments() []string {
+		var normalizedElement string
+		normalizedElement, err = Normalize(e)
+		if err != nil {
+			return
```
So, if normalizing one out of potential multiple arguments fails, the others after that argument are also not part of the arguments anymore.. I don't like this behavior. I would expect that either I get an empty array in error case or I get all elements and the ones that failed are not normalized or I get all arguments except the ones that couldn't be normalized.

kiliankoe replied:
I would've expected to check the error on the other side and if present, ignore the partially normalized input altogether. But you're right, people can also choose to ignore the error and will get stumped as to why they're only getting a part of the results.

BenchR267 replied:
Yeah you are right, but let's make it also robust enough for people that don't do error handling :D I changed it in 1735ad2 to use continue, so that not normalized elements are now not part of the result anymore.

Commit:
Using continue instead of return

BenchR267 commented:
Thanks! 👌

kiliankoe commented:
Shouldn't @GoalfredBot comment again? Or am I misunderstanding how danger works? :D

BenchR267 commented:
Yes, he should. His message on the CI was 'All green. Congrats.'
Maybe it's an option if he posts green builds^^
